src/screens/TokenSwap.js

```diff
@@ -0,0 +1,394 @@
+
+  const [swapDirection, setSwapDirection] = useState(null);
+
+  // XXX: this is a mock value to simulate fetching and showing the quote section
+  const [showQuote, setShowQuote] = useState(false);
+
+  const {
+    allowedTokens,
+    inputToken,
+    outputToken,
+  } = useSelector((state) => state.tokenSwap);
+  const { decimalPlaces } = useSelector((state) => ({
+    decimalPlaces: state.serverInfo?.decimal_places
+  }));
+
+  const navigation = useNavigation();
+
+  useEffect(() => {
+    if (!inputToken) {
+      dispatch(tokenSwapSetInputToken(allowedTokens[0]));
+      return;
+    }
+  }, [inputToken]);
+
+  useEffect(() => {
+    if (!outputToken) {
+      dispatch(tokenSwapSetOutputToken(allowedTokens[0]));
+      return;
+    }
+  }, [outputToken]);
+
+  function onInputAmountChange(text, value) {
+    setInputTokenAmountStr(text);
+    setInputTokenAmount(value);
+  }
+
+  function onInputAmountEndEditing(_target) {
+    console.log(`User added ${inputTokenAmount} tokens on input`);
+    setOutputTokenAmountStr('0');
+    setOutputTokenAmount(0n);
+    setSwapDirection('input');
+
+    // XXX here we should fetch the quote, but we will mock for now
+    setShowQuote(true);
+    setOutputTokenAmountStr('6.41');
+    setOutputTokenAmount(641n);
+  }
+
+  function onOutputAmountEndEditing(_target) {
+    console.log(`User added ${outputTokenAmount} tokens on output`);
+    setInputTokenAmountStr('0');
+    setInputTokenAmount(0n);
+    setSwapDirection('output');
+
+    // XXX here we should fetch the quote, but we will mock for now
+    setShowQuote(true);
+    setInputTokenAmountStr('6.41');
+    setInputTokenAmount(641n);
+  }
+
+  function onOutputAmountChange(text, value) {
+    setOutputTokenAmountStr(text);
+    setOutputTokenAmount(value);
+  }
+
+  function onFocus() {
+    setShowQuote(false);
+  }
+
+  const onInputTokenBoxPress = () => {
+    navigation.navigate('TokenSwapListInputToken', { token: inputToken });
+  };
+
+  const onOutputTokenBoxPress = () => {
+    navigation.navigate('TokenSwapListOutputToken', { token: outputToken });
+  };
+
+  const switchTokens = () => {
+
+    const newOutputStr = inputTokenAmountStr;
+    const newOutput = inputTokenAmount;
+    const newInputStr = outputTokenAmountStr;
+    const newInput = outputTokenAmount;
+
+    // Also switch the direction of the swap
+    if (swapDirection === 'input') {
+      setInputTokenAmountStr(newInputStr);
+      setInputTokenAmount(newInput);
+
+      setOutputTokenAmountStr('0');
+      setOutputTokenAmount(0n);
+
+      setSwapDirection('output');
+    } else if (swapDirection === 'output') {
+      setOutputTokenAmountStr(newOutputStr);
+      setOutputTokenAmount(newOutput);
+
+      setInputTokenAmountStr('0');
+      setInputTokenAmount(0n);
+
+      setSwapDirection('input');
+    }
+
+    // switch the tokens being swapped
+    dispatch(tokenSwapSwitchTokens());
+  };
+
+  const isButtonDisabled = () => {
```

Comment:
chore: Specify which button for clarity
```suggestion
  const isReviewButtonDisabled = () => {
```